compiler/rustc_parse/src/parser/stmt.rs

```diff
@@ -585,7 +591,11 @@ impl<'a> Parser<'a> {
                         e.emit();
                         self.recover_stmt();
                     }
-                    // Don't complain about type errors in body tail after parse error (#57383).
+                    skip_type_error = true;
+                }
+
+                if skip_type_error {
```

Comment:
Can you give this a different name? `skip_type_error` -> `replace_with_err` or something?